bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableIO.java

```diff
@@ -1034,15 +1037,20 @@ public void populateDisplayData(DisplayData.Builder builder) {
    * org.apache.hadoop.hbase.client.Append}s and {@link org.apache.hadoop.hbase.client.Increment}s.
    * This limitation exists because if the batch fails partway through, Appends/Increments might be
    * re-run, causing the {@link Mutation} to be executed twice, which is never the user's intent.
-   * Re-running a Delete will not cause any differences.  Re-running a Put isn't normally a problem,
+   * Re-running a Delete will not cause any differences. Re-running a Put isn't normally a problem,
    * but might cause problems in some cases when the number of versions supported by the column
-   * family is greater than one.  In a case where multiple versions could be a problem, it's best to
+   * family is greater than one. In a case where multiple versions could be a problem, it's best to
    * add a timestamp to the {@link Put}.
    */
-   public static PTransform<PCollection<KV<String, Iterable<Mutation>>>, PDone>
-      writeToMultipleTables(CloudBigtableConfiguration config) {
-    validateConfig(config);
-    return new CloudBigtableWriteTransform<>(new CloudBigtableMultiTableWriteFn(config));
+  public static PTransform<PCollection<KV<String, Iterable<Mutation>>>, PDone>
+      writeToMultipleTables(final CloudBigtableConfiguration config) {
+    return new CloudBigtableWriteTransform<KV<String, Iterable<Mutation>>>(
+        new CloudBigtableMultiTableWriteFn(config)) {
+      @Override
+      public void validate(PipelineOptions options) {
```

Comment:
Can you please put the validation code directly in the `CloudBigtableWriteTransform` class (it would also mean passing the config in the constructor)?

Reply:
Putting the validation code in CloudBigtableWriteTransform seems to the
right approach. There were two reasons I didn't end up doing that:
1. The constructor is public. I am not sure if it should be.
2. There are two different config classes. I guess we can call Validate ()
on the config class if it is defined.

Reply:
BTW, I made CloudBigtableWriteTransform private and passed the config through the constructor.
I also updated config classes to include validate() method.